### PR TITLE
Tide branch protection warnings turn to debug

### PR DESCRIPTION
**Pull request by chaodaiG**
These are pretty much intended behavior from users, not worth log as warnings

Fixes https://github.com/kubernetes/test-infra/issues/21983

### CODE REVIEW

File: prow/config/branch_protection.go [hunk 1]
```diff
@@ -444,11 +444,10 @@ func (c *Config) unprotectedBranches(presubmits map[string][]Presubmit) []string
 //     because any branches not explicitly specified in the configuration will be unprotected.
 func (c *Config) BranchProtectionWarnings(logger *logrus.Entry, presubmits map[string][]Presubmit) {
 	if warnings := c.reposWithDisabledPolicy(); len(warnings) > 0 {
-
-		logger.WithField("repos", strings.Join(warnings, ",")).Warn("The following repos define a policy, but have protect: false")
```
Comment (alvaroaleman) [inline prow/config/branch_protection.go:448]: Are you sure? AFAIK those are essentially misconfigurations
/cc @petr-muller

Comment (chaodaiG) [reply prow/config/branch_protection.go:448]: yeah I believe so, see https://github.com/kubernetes/test-infra/issues/21983#issuecomment-827969398

Comment (petr-muller) [reply prow/config/branch_protection.go:448]: I think the PR is correct, I think "policy" here is usually inferred from the presence of required jobs, and the fact that you have required jobs does not always mean you want to manage BP.

**Pull request merged**